thunder/core/proxies.py

```diff
@@ -1316,6 +1316,17 @@ def __getattr__(self, attr: str, /):
 
         return method_or_value
 
+    def __iter__(self):
+        # NOTE: this implementation is equivalent to torch.Tensor.__iter__
+
+        if self.ndim == 0:
+            raise TypeError("iteration over a 0-dim tensor")
+
+        import thunder.torch
+
+        unbound_tuple = thunder.torch.unbind(self)
+        return iter(unbound_tuple)
```

Comment:
I think we can also do `iter(self.unbind()` as `unbind` is also a registered method.

https://github.com/Lightning-AI/lightning-thunder/blob/0d804443b2d499d897d0e5cb3cc96d508ec2454a/thunder/torch/__init__.py#L1256-L1262

Reply:
Hmm, did not know that things changed with the API :)

Reply:
Applied! Thank you!